mne/io/brainvision/tests/test_brainvision.py

```diff
@@ -77,82 +77,50 @@ def test_orig_units(recwarn):
-    tmp_vmrk_file = tmp_vhdr_file.replace('.vhdr', '.vmrk')
-    tmp_eeg_file = op.join(tmpdir, op.basename(eeg_path))
-    sh.copyfile(vhdr_bad_date, tmp_vhdr_file)
-    sh.copyfile(vhdr_bad_date.replace('.vhdr', '.vmrk'), tmp_vmrk_file)
-    sh.copyfile(eeg_path, tmp_eeg_file)
-
-    # We'll exclusively manipulate the "New Segment" line in vmrk
-    with open(tmp_vmrk_file, 'r') as fin:
-        lines = fin.readlines()
-    idx = lines.index('Mk1=New Segment,,1,1,0,00000000000304125000\n')
+    return tmp_vhdr_file, tmp_vmrk_file, lines, idx
 
-    # Now perform some tests
-    # Test that we get no error for trying to extract meas_date if there is
-    # no marker of type "New Segment" in the data
-    lines[idx] = 'Mk1=STATUS,,1,1,0\n'
-    with open(tmp_vmrk_file, 'w') as fout:
-        fout.writelines(lines)
-    raw = read_raw_brainvision(tmp_vhdr_file)
-    assert raw.info['meas_date'] is None
-    assert 'unspecified' in repr(raw.info)
 
-    # Test that we extract no date, if "New Segment", but no date specified
-    # Note the trailing comma but missing data following that comma
-    lines[idx] = 'Mk1=New Segment,,1,1,0,\n'
-    with open(tmp_vmrk_file, 'w') as fout:
-        fout.writelines(lines)
-    raw = read_raw_brainvision(tmp_vhdr_file)
-    assert raw.info['meas_date'] is None
-    assert 'unspecified' in repr(raw.info)
+@pytest.mark.parametrize('newstring, expected', [
+    pytest.param('Mk1=New Segment,,1,1,0,20131113161403794232\n', '2013-11-13 16:14:03 GMT'),  # noqa: E501
+    pytest.param('Mk1=New Segment,,1,1,0,20070716122240937454\nMk2=New Segment,,2,1,0,20070716122240937455\n', '2007-07-16 12:22:40 GMT'),  # noqa: E501
+    pytest.param('Mk1=STATUS,,1,1,0\n', 'unspecified'),
+    pytest.param('Mk1=New Segment,,1,1,0,\n', 'unspecified'),
+    pytest.param('Mk1=New Segment,,1,1,0\n', 'unspecified'),
+    pytest.param('Mk1=New Segment,,1,1,0,00000000000304125000', 'unspecified'),
+    pytest.param('Mk1=New Segment,,1,1,0,\nMk2=New Segment,,2,1,0,20070716122240937454\n', '2007-07-16 12:22:40 GMT'),  # noqa: E501
+])
+def test_dates(_get_date_test_data, newstring, expected):
+    """Test valid dates."""
+    tmp_vhdr_file, tmp_vmrk_file, lines, idx = _get_date_test_data
 
-    # Test that no error if "New Segment", but no date specified and no
-    # trailing comma
-    lines[idx] = 'Mk1=New Segment,,1,1,0\n'
-    with open(tmp_vmrk_file, 'w') as fout:
-        fout.writelines(lines)
-    raw = read_raw_brainvision(tmp_vhdr_file)
-    assert raw.info['meas_date'] is None
-    assert 'unspecified' in repr(raw.info)
+    # Replace the New Segment
+    lines[idx] = newstring
 
-    # Test that a fine date gets recognized, use 2000-01-01:12:00:00, assuming
-    # UTC timezone, which would be in unix time: 946728000
-    lines[idx] = 'Mk1=New Segment,,1,1,0,20000101120000000000\n'
+    # Write it to tmp file
     with open(tmp_vmrk_file, 'w') as fout:
         fout.writelines(lines)
```

Comment:
I think I got it.
The fixture should use `tmpdir_factory`, the writing of the file should be done inside the fixture and the fixture should `yield file_name, intermediate_rep, date_repr`

See https://github.com/pytest-dev/pytest/issues/1595

Reply:
check yield fixtures here https://docs.pytest.org/en/latest/fixture.html#fixtures